src/lib/jwk-store.ts

```diff
@@ -19,71 +19,148 @@
+  RS384: RsaPrivateFieldsRemover,
+  RS512: RsaPrivateFieldsRemover,
+
+  // RSASSA-PSS
+  PS256: RsaPrivateFieldsRemover,
+  PS384: RsaPrivateFieldsRemover,
+  PS512: RsaPrivateFieldsRemover,
+
+  // ECDSA
+  ES256: EcdsaPrivateFieldsRemover,
+  ES256K: EcdsaPrivateFieldsRemover,
+  ES384: EcdsaPrivateFieldsRemover,
+  ES512: EcdsaPrivateFieldsRemover,
+
+  // Edwards-curve DSA
+  EdDSA: EddsaPrivateFieldsRemover,
+};
+
+const supportedAlgs = Object.keys(privateToPublicTransformerMap);
+
+const normalizeKey = (jwk: JWK, opts?: { kid?: string }): void => {
+  if (jwk.kid !== undefined) {
+    return;
+  }
+
+  if (opts !== undefined && opts.kid !== undefined) {
+    jwk.kid = opts.kid;
+  } else {
+    jwk.kid = generateRandomKid();
+  }
+};
 
 /**
- * Simplified wrapper class for [node-jose]{@link https://github.com/cisco/node-jose}'s keystore.
+ * Simple JWK store
  */
 export class JWKStore {
-  #store: JWK.KeyStore;
   #keyRotator: KeyRotator;
 
   /**
    * Creates a new instance of the keystore.
    */
   constructor() {
-    this.#store = JWK.createKeyStore();
     this.#keyRotator = new KeyRotator();
   }
 
   /**
-   * Generates a new random RSA key and adds it into this keystore.
+   * Generates a new random key and adds it into this keystore.
    *
-   * @param {number} [size] The size in bits of the new key. Default: 2048.
-   * @param {string} [kid] The key ID. If omitted, a new random 'kid' will be generated.
-   * @param {string} [use] The intended use of the key (e.g. 'sig', 'enc'.) Default: 'sig'.
-   * @returns {Promise<JWK.Key>} The promise for the generated key.
+   * @param {string} alg The selected algorithm.
+   * @param {object} [opts] The options.
+   * @param {string} [opts.kid] The key identifier to use.
+   * @param {string} [opts.crv] The OKP "crv" to be used for "EdDSA" algorithm.
+   * @returns {Promise<JWK>} The promise for the generated key.
    */
-  async generateRSA(
-    size?: number,
-    kid?: string,
-    use = 'sig'
-  ): Promise<JWK.Key> {
-    const key = await this.#store.generate('RSA', size, { kid, use });
-    this.#keyRotator.add(key);
-    return key;
+  async generate(
+    alg: string,
+    opts?: { kid?: string; crv?: string }
+  ): Promise<JWK> {
+    const generateOpts: GenerateKeyPairOptions =
+      opts !== undefined && opts.crv !== undefined ? { crv: opts.crv } : {};
+
+    const pair = await generateKeyPair(alg, generateOpts);
+    const jwk = await fromKeyLike(pair.privateKey);
+
+    jwk.alg = alg;
+    normalizeKey(jwk, opts);
+
+    this.#keyRotator.add(jwk);
+    return jwk;
   }
 
   /**
    * Adds a JWK key to this keystore.
    *
-   * @param {JWK.Key} jwk The JWK key to add.
-   * @returns {Promise<JWK.Key>} The promise for the added key.
+   * @param {JWK} jwk The JWK key to add.
+   * @returns {Promise<JWK>} The promise for the added key.
    */
-  async add(jwk: JWK.Key): Promise<JWK.Key> {
-    const jwkUse: JWK.Key = { ...jwk };
+  async add(jwk: JWK): Promise<JWK> {
```

Comment:
Maybe change the signature to accept a Record<string, unknown> and return a JWK where alg, kid are always specified
This would allow us to export our own JWK type (only extending jose's one)

Reply:
Fixed